remotecommanddispatcher/src/main/java/com/tealium/remotecommanddispatcher/RemoteCommandDispatcher.kt

```diff
@@ -21,31 +20,14 @@ interface RemoteCommandDispatcherListener : DispatcherListener {
  * Tag Management or JSON-controlled
  */
 class RemoteCommandDispatcher(private val context: TealiumContext,
-                              private val afterDispatchSendCallbacks: AfterDispatchSendCallbacks,
-                              private val client: NetworkClient = HttpClient(context.config)) : Dispatcher, RemoteCommandListener {
-
-    private val webViewCommands = mutableMapOf<String, RemoteCommand>()
-    private val jsonCommands = mutableMapOf<String, RemoteCommand>()
+                              private val client: NetworkClient = HttpClient(context.config),
+                              private val manager: CommandsManager = RemoteCommandsManager(context.config)) : Dispatcher, RemoteCommandListener {
 
     /**
      * Adds remote commands to be evaluated when triggered
      */
-    fun add(remoteCommand: RemoteCommand) {
-        when (remoteCommand) {
-            is JsonRemoteCommand -> {
-                jsonCommands[remoteCommand.commandName] = remoteCommand
-                remoteCommand.filename?.let {
-                    remoteCommand.remoteCommandConfigRetriever = RemoteCommandConfigRetriever(context.config, remoteCommand.commandName, filename = it)
-                } ?: run {
-                    remoteCommand.remoteUrl?.let {
-                        remoteCommand.remoteCommandConfigRetriever = RemoteCommandConfigRetriever(context.config, remoteCommand.commandName, remoteUrl = it)
-                    } ?: run {
-                        Logger.dev(BuildConfig.TAG, "No filename or remote url found for JSON Remote command: ${remoteCommand.commandName}")
-                    }
-                }
-            }
-            else -> webViewCommands[remoteCommand.commandName] = remoteCommand
-        }
+    fun add(remoteCommand: RemoteCommand, filename: String? = null, remoteUrl: String? = null) {
```

Comment:
being picky, but it may be useful to cover the input options here in the method docs - as the optional parameters aren't necessarily self-explaining. For instance, do you need to supply both? etc